### ISSUE TITLE
HREBSDMain parfor, ImageInd, dissimilar structures

### ISSUE THREAD

**Issue by caldrews** [BYU-MicrostructureOfMaterials/OpenXY#26]
I am continuing to face some issue within HREBSDMain with parallel processing. I am receiving an error at line 66, parfor(ImageInd = indVect,NumberOfCores). I have tried preallocating the ImageInd variable before the parfor loop in an attempt to fix this, to no avail. I have downloaded a fresh copy of OpenXY and the issue persists. I am using MATLAB 2018b and will attempt on 2020 on a different computer and update with results.

Any idea what could be triggering this error? With the issues encountered with fitMetrics and now this, is there value in rolling back to an older version?
![image](https://user-images.githubusercontent.com/79323764/109436300-d94ee080-79ec-11eb-848c-489278420351.png)

**Comment by eretnek** [comment]
Matlab often has trouble finding the problematic line in a parfor loop, so
it just calls out the error on the loop itself. What changes have you made
to get this error? It looks like another version of the wrong shape error.
You can try running on a single processor to more easily track down the
error.

My guess is that this poor error handling goes back years, so reverting
might be difficult, but you're welcome to try. I can send you a version
that solves your first error, but I dont think I have enough info to
duplicate the error you mention in this email.

**Comment by caldrews** [comment]
Thanks,

I think the only changes I've made would be in the GetEMSoftDir function, to just have it point at the directory instead of scanning the config file. I tried with a fresh installation, however we have multiple users and multiple openXY folders/versions, so I'd have to see if at some point its calling out to a different function/version. I'll reach out to you via email about the older version, but I think you're correct now that I've tinkered with it. Running the same code on my own desktop, with MATLAB 2020, 3 working processors instead of 11 I get no error, so it is likely on my end and not a problem with the code itself.

I'll go ahead and close this out, I don't think its an OpenXY problem.

**Comment by caldrews** [comment]
Thanks,

I think the only changes I've made would be in the GetEMSoftDir function, to just have it point at the directory instead of scanning the config file. I tried with a fresh installation, however we have multiple users and multiple openXY folders/versions, so I'd have to see if at some point its calling out to a different function/version. I'll reach out to you via email about the older version, but I think you're correct now that I've tinkered with it. Running the same code on my own desktop, with MATLAB 2020, 3 working processors instead of 11 I get no error, so it is likely on my end and not a problem with the code itself.

I'll go ahead and close this out, I don't think its an OpenXY problem.
